### PR TITLE
PRC-328: API to get prisoner contact restrictions

### PR DESCRIPTION
**Pull request by james-baker-moj**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppscontactsapi/resource/PrisonerContactController.kt [hunk 1]
```diff
@@ -145,4 +148,48 @@ class PrisonerContactController(
...
+        ],
+      ),
+    ],
+  )
+  @GetMapping(value = ["/{prisonerContactId}/restrictions"], produces = [MediaType.APPLICATION_JSON_VALUE])
+  @PreAuthorize("hasAnyRole('ROLE_CONTACTS_ADMIN')")
+  fun getPrisonerContactRestrictionsByPrisonerContactId(
+    @PathVariable("prisonerContactId") @Parameter(
+      name = "prisonerContactId",
+      description = "The id of the prisoner contact relationship to get relevant restrictions for",
```
Comment (prabash-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppscontactsapi/resource/PrisonerContactController.kt:191]: is this description not confusing,? should it be `the prisoner contact id to get the relevent restrictions`

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppscontactsapi/service/RestrictionsService.kt [hunk 2]
```diff
@@ -32,6 +38,32 @@ class RestrictionsService(
     }
   }
 
+  fun getPrisonerContactRestrictions(prisonerContactId: Long): PrisonerContactRestrictionsResponse {
+    val prisonerContact = prisonerContactRepository.findById(prisonerContactId)
+      .orElseThrow { EntityNotFoundException("Prisoner contact relationship ($prisonerContactId) could not be found") }
```
Comment (prabash-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppscontactsapi/service/RestrictionsService.kt:43]: should it not be "Prisoner contact ($prisonerContactId) could not be found?"

File: src/test/kotlin/uk/gov/justice/digital/hmpps/hmppscontactsapi/resource/PrisonerContactControllerTest.kt [hunk 3]
```diff
@@ -127,4 +130,28 @@ class PrisonerContactControllerTest {
       }
     }
   }
+
+  @Nested
+  inner class GetPrisonerContactRelationshipRestrictions {
+    private val prisonerContactId = 123456L
+
+    @Test
+    fun `should create a contact relationship successfully`() {
```
Comment (prabash-moj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/hmppscontactsapi/resource/PrisonerContactControllerTest.kt:139]: `should get a contact relationship successfully`?

**Review by prabash-moj — APPROVED** [review]

**Review by timharrison-moj — APPROVED** [review]
LGTM. I have some changes coming to the sync are for contact restrictions but I don't think we've overlapped anywhere.

**Pull request merged**